triton-vm/src/table/processor_table.rs

```diff
@@ -907,28 +1131,36 @@ impl ExtProcessorTable {
         // constraints related to evaluation and permutation arguments
 
         transition_constraints
-            .push(factory.running_evaluation_for_standard_input_updates_correctly(challenges));
-        transition_constraints
-            .push(factory.running_product_for_instruction_table_updates_correctly(challenges));
+            .push(factory.running_evaluation_for_standard_input_updates_correctly());
         transition_constraints
-            .push(factory.running_evaluation_for_standard_output_updates_correctly(challenges));
+            .push(factory.running_product_for_instruction_table_updates_correctly());
         transition_constraints
-            .push(factory.running_product_for_op_stack_table_updates_correctly(challenges));
+            .push(factory.running_evaluation_for_standard_output_updates_correctly());
+        transition_constraints.push(factory.running_product_for_op_stack_table_updates_correctly());
+        transition_constraints.push(factory.running_product_for_ram_table_updates_correctly());
         transition_constraints
-            .push(factory.running_product_for_ram_table_updates_correctly(challenges));
-        transition_constraints
-            .push(factory.running_product_for_jump_stack_table_updates_correctly(challenges));
-        transition_constraints
-            .push(factory.running_evaluation_to_hash_table_updates_correctly(challenges));
+            .push(factory.running_product_for_jump_stack_table_updates_correctly());
+        transition_constraints.push(factory.running_evaluation_to_hash_table_updates_correctly());
+        transition_constraints.push(factory.running_evaluation_from_hash_table_updates_correctly());
+
+        // TODO: Consider folding constants here
+
         transition_constraints
-            .push(factory.running_evaluation_from_hash_table_updates_correctly(challenges));
+            .into_iter()
+            .map(|tc_ref| tc_ref.consume())
+            .collect_vec()
+    }
 
-        // normalize polynomials to speed up evaluation
-        for polynomial in transition_constraints.iter_mut() {
-            polynomial.normalize();
+    fn ext_transition_constraints(
+        challenges: &ProcessorTableChallenges,
+    ) -> Vec<MPolynomial<XFieldElement>> {
+        let circuits = Self::ext_transition_constraints_as_circuits();
+        let mut ret: Vec<MPolynomial<XFieldElement>> = vec![];
+        for circuit in circuits {
+            ret.push(circuit.partial_evaluate(challenges));
         }
 
-        transition_constraints
+        ret
```

Comment:
Consider onelining; perhaps something along the lines of
```
circuits.into_iter().map(|circ| circ.partial_evaluate(challenges)).collect_vec()
```

Reply:
OK